File: Android/Pittsburgh_Realtime_Tracker/app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/world/jsonpojo/Pt.java

package rectangledbmi.com.pittsburghrealtimetracker.world.jsonpojo;

import javax.annotation.Generated;
import com.google.gson.annotations.Expose;

@Generated("org.jsonschema2pojo")
public class Pt {

    @Expose
    private int seq;
    @Expose
    private double lat;
    @Expose
    private double lon;
    @Expose
    private char typ;
    @Expose
    private int stpid;
    @Expose
    private String stpnm;
    @Expose
    private double pdist;
    @Expose
    private String msg;

    /**
     *
     * @return
     *     The seq
     */
    public int getSeq() {
        return seq;
    }

    /**
     *
     * @param seq
     *     The seq
     */
    public void setSeq(int seq) {
        this.seq = seq;
    }

    /**
     *
     * @return
     *     The lat
     */
    public double getLat() {
        return lat;
    }

    /**
     *
     * @param lat
     *     The lat
     */
    public void setLat(double lat) {
        this.lat = lat;
    }

    /**
     *
     * @return
     *     The lon
     */
    public double getLon() {
        return lon;
    }

    /**
     *
     * @param lon
     *     The lon
     */
    public void setLon(double lon) {
        this.lon = lon;
    }

    /**
     *
     * @return
     *     The typ
     */
    public char getTyp() {
        return typ;
    }

    /**
     *
     * @param typ
     *     The typ
     */
    public void setTyp(String typ) {
        this.typ = typ.charAt(0);
    }

    /**
     *
     * @return
     *     The stpid
     */
    public int getStpid() {
        return stpid;
    }

    /**
     *
     * @param stpid
     *     The stpid
     */
    public void setStpid(String stpid) {
        this.stpid = Integer.parseInt(stpid);
    }

    /**
     *
     * @return
     *     The stpnm
     */
    public String getStpnm() {
        return stpnm;
    }

    /**
     *
     * @param stpnm
     *     The stpnm
     */
    public void setStpnm(String stpnm) {
        this.stpnm = stpnm;
    }

    /**
     *
     * @return
     *     The pdist
     */
    public double getPdist() {
        return pdist;
    }

    /**
     *
     * @param pdist
     *     The pdist
     */
    public void setPdist(double pdist) {
        this.pdist = pdist;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

}

File: pat-static/src/main/java/com/rectanglel/patstatic/patterns/response/Pt.kt
package com.rectanglel.patstatic.patterns.response

import javax.annotation.processing.Generated
import com.google.gson.annotations.Expose
import com.rectanglel.patstatic.predictions.PredictionsType

/**
 * object for each point in a pattern
 *
 * @author Jeremy Jao
 * @since 46
 */
@Generated("org.jsonschema2pojo")
data class Pt(@Expose val seq: Int,
              @Expose val lat: Double = 0.toDouble(),
              @Expose val lon: Double = 0.toDouble(),
              @Expose val typ: Char = ' ',
              @Expose val stpid: Int = 0,
              @Expose val stpnm: String? = null,
              @Expose val pdist: Double = 0.toDouble(),
              @Expose val msg: String? = null,
              @Expose var rtdir: String? = null) : PredictionsType {

    override val id: Int
        get() = if (typ == 'S') {
            stpid
        } else -1

    override val title: String
        get() = String.format("(%d) %s - %s", stpid, stpnm, rtdir)
}
